pwndbg/disasm/x86.py

```diff
@@ -431,6 +431,28 @@ def _condition(self, instruction: PwndbgInstruction, emu: Emulator) -> Instructi
 
         return InstructionCondition.TRUE if bool(conditional) else InstructionCondition.FALSE
 
+    # Override
+    def _get_syscall_arch(self, instruction: PwndbgInstruction) -> str | None:
+        syscall_arch = pwndbg.gdblib.arch.name
```

Comment:
Random thought: do we really have to fetch this at this point if we know we are in x86? :p

Reply:
Correct - this class supports both x86 and x86_64. I added a note about this in this function as well as the before the class declaration.